internal/graph/storagewrapper.go

```diff
@@ -232,26 +232,30 @@ func (c *CachedDatastore) Read(
 	return c.newCachedIteratorByObjectRelation(ctx, "Read", store, iter, b.String(), tupleKey.GetObject(), tupleKey.GetRelation())
 }
 
-func (c *CachedDatastore) findInCache(store, key string, invalidEntityKeys []string) (*storage.TupleIteratorCacheEntry, bool) {
+func findInCache(cache storage.InMemoryCache[any], store, key string, invalidEntityKeys []string) (*storage.TupleIteratorCacheEntry, bool) {
 	var tupleEntry *storage.TupleIteratorCacheEntry
+	var ok bool
 
 	// The iterator cache has a TTL and will eventually consistent.
-	if res := c.cache.Get(key); res != nil {
-		tupleEntry = res.(*storage.TupleIteratorCacheEntry)
+	if res := cache.Get(key); res != nil {
+		tupleEntry, ok = res.(*storage.TupleIteratorCacheEntry)
```

Comment:
we actually should validate the tupleEntry to ensure we are are in a sane state and not-panic.  Better to invalidate cache then panic / die